Organize repo

adamaltman opened the pull request:
- [ ] abstract Session.
- [ ] SRP for models (make entities + sql mapper classes).
- [ ] views

adamaltman commented:
@brandonsavage test tag

brandonsavage commented on composer.json:
```diff
@@ -1,5 +1,11 @@
+    },
+    "require": {
+        "aura/di": "^2.2",
+        "nikic/fast-route": "^0.7.0"
```
I assume this is the router you're going to implement?

adamaltman replied:
Yes.  If I have time I may try and replace it with my own.

brandonsavage commented on diconfig.php:
```diff
@@ -0,0 +1,81 @@
+    'dsn' => $dsn,
+    'username' => $db['user'],
+    'passwd' => $db['pass'],
+    'options' => [
```
Good catch. :smile:

brandonsavage commented on public/index.php:
```diff
@@ -3,12 +3,9 @@
-$framework = new MasterController($config);
-echo $framework->execute();
\ No newline at end of file
+$framework = new Masterclass\MasterController($di, $config);
```
I'm kind of surprised not to see the router injected here.

adamaltman replied:
I did that originally, but then changed it to be injected into `RouteMap`.  This was so that I could use the DI config to build the routes.  Otherwise, I had built them in a separate file `routes.php` and included in the `index.php`.  

I could still do that... maybe it would be better.

brandonsavage commented on src/Controller/Comment.php:
```diff
@@ -0,0 +1,36 @@
+        $this->commentModel->addComment(
+            $_SESSION['username'],
+            $this->request->getPostParam('story_id'),
+            filter_input(INPUT_POST, 'comment', FILTER_SANITIZE_FULL_SPECIAL_CHARS)
```
We're injecting a request object, so we now have an opportunity to make use of it.

brandonsavage commented on src/Controller/Comment.php:
```diff
@@ -0,0 +1,36 @@
+            filter_input(INPUT_POST, 'comment', FILTER_SANITIZE_FULL_SPECIAL_CHARS)
+        );
+
+        header("Location: /story/?id=" . $_POST['story_id']);
```
Same as above.

brandonsavage commented on src/Controller/Index.php:
```diff
@@ -0,0 +1,38 @@
+        
+        $content .= '</ol>';
+        
+        require '/vagrant/layout.phtml';
```
I would caution against hard-coding this, because you're not likely to deploy on vagrant.

brandonsavage commented on src/Controller/Story.php:
```diff
@@ -0,0 +1,101 @@
+
+    public function index()
+    {
+        $id = $this->request->getQueryParam('id');
```
Good! :)

brandonsavage commented on src/Controller/Story.php:
```diff
@@ -0,0 +1,101 @@
+
+        $error = '';
+        if (isset($_POST['create'])) {
+            if (!isset($_POST['headline']) || !isset($_POST['url']) ||
```
Again, we have access to the request now.

brandonsavage commented on src/Controller/User.php:
```diff
@@ -0,0 +1,160 @@
+        if ($this->request->getPostParam('create')) {
+
+            $username = $this->request->getPostParam('username');
+            $password = $this->request->getPostParam('password'); $_POST['password'];
```
The $_POST['password'] is superfluous at this point.

brandonsavage commented on src/Model/Story.php:
```diff
@@ -0,0 +1,55 @@
+{
+    protected $dataStore;
+
+    public function __construct(DataStore $dataStore)
```
You've implemented ActiveRecord. :-D

brandonsavage commented on src/Model/User.php:
```diff
@@ -0,0 +1,110 @@
+    protected $email;
+    protected $hashedPassword;
+
+    public function __construct(DataStore $dataStore)
```
You've implemented ActiveRecord. :-D

adamaltman replied:
Yes.  Haha.  I have it in my todo to adjust that.

brandonsavage commented on src/MasterController.php:
```diff
@@ -0,0 +1,31 @@
+
+    public function execute()
+    {
+        $routeMap = $this->container->newInstance(RouteMap::class);
```
Ah I see what you did here, Typically I'd advocate for injecting as much as you possibly can. We can't mock this, because I'm ALWAYS going to get an instance of RouteMap::class. I'd suggest injecting this in the __construct().

brandonsavage commented:
Step in the right direction and a good start. You still have some work to do but looking good!

adamaltman commented:
Still much to do.... I made domain models, but they aren't in use yet. (And I didn't quite finish it... I didn't realize how long this would take...)

brandonsavage commented:
Adam, I'm going to recommend that after I review a PR that you merge it, so
that I can more easily help keep track of what you've changed. Right now
I'm not sure what you've changed here.

On Wed, Mar 16, 2016 at 1:25 AM, Adam Altman notifications@github.com
wrote:

> Still much to do.... I made domain models, but they aren't in use yet.
> (And I didn't quite finish it... I didn't realize how long this would
> take...)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/adamaltman/masterclass-repo/pull/2#issuecomment-197158818

adamaltman commented:
Ahh, yes, I changed two main things: pass the route destination into the MasterController.  And I added the "Domain" folder (and files within there).

adamaltman commented:
I'll merge it before I do any more changes.
